### ISSUE TITLE
python-zstandard incompatible with pypy

### ISSUE THREAD

**Issue by c-wicklein** [indygreg/python-zstandard#31]
Greetings,

I ran into another issue when I started using python-zstandard with pypy.  The module builds, but it fails at runtime because the bytes objects can't be resized.  My workaround has been to change decompress2 to return a list of strings (iterate over -- decompression, PyBytes_FromStringAndSize on the output, PyList_Append with the result, and Py_DECREF the result) instead of growing a single string.  I suspect this change makes things slower (not to mention incompatible), but it was worth it for me to get pypy working quickly.

**Comment by c-wicklein** [comment]
For reference, I've attached here the complete c-ext/decompressobj.c I'm now using.  It's based on 0.8.1, has the original decompress method restored, and includes my decompress2 method which now returns a list of strings along with an index to the beginning of the unused input data.  This addresses both the performance problem associated with unknown framing and compatibility with pypy.

[decompressobj.c.txt](https://github.com/indygreg/python-zstandard/files/1288808/decompressobj.c.txt)

**Comment by indygreg** [comment]
I encountered this failure as well. I /think/ the problem is due to `_PyBytes_Resize()` not working with bytes of size 1. And the reason that doesn't work is because a 1 sized PyBytes isn't a real PyObject!

If this was the full extent of the bug, then I fixed this in `master` a few minutes ago. There may be other open issues with `_PyBytes_Resize()` in the code base though. I need to audit for that.
